parser_library/src/workspaces/workspace_configuration.cpp

```diff
@@ -359,7 +362,7 @@ utils::task workspace_configuration::process_processor_group_library(const confi
     if (!lib_path.has_value())
     {
         diags.push_back(diagnostic_s::warning_L0006(m_proc_grps_loc, lib.path));
-        co_return;
+        return {};
```

Comment:
It almost feels like emscripten doesn't like "implicit" return types from `co_return` and we need to explicitly specify `return {}`. However, `co_return` can still be used elsewhere....

How do you find these kinds of issues? Just thorough testing and luck? Can we automate the detection process somehow?

Reply:
Looking just a few lines below, the case with `search_root` is even weirder to me

Reply:
The `search_root`, that's classic undefined order of argument evaluation...

Reply:
As for the `co_return` and `return {}` - this change means, that the function is not a coroutine (there are no `co_return`, `co_await` or `co_yield` keywords). It is just a regular function that happens to return a `task` object.

So the caller will get an empty task and just continue processing and most importantly it won't call co_await on it, otherwise the emscripten/llvm would generate a function call instead of the symmetric transfer - which is the root of the problem.

As a side effect, it is more efficient, since we will allocate coroutine frame only when we actually might need it.